Added ADuM5410, ADuM5411 and ADuM5412

luca-della-vedova opened the pull request:
Added Analog Devices new ADuM451x series isolators.
------------
Thanks for creating a pull request to contribute to the KiCad libraries! To speed up integration of your PR, please check the following items:

- [ ] Provide a URL to a datasheet for the symbol(s) you are contributing
https://www.analog.com/media/en/technical-documentation/data-sheets/ADuM5410-5411-5412.pdf
- [x] An example screenshot image is very helpful
Example from the 4512 series (2 inputs and 2 outputs)
![image](https://user-images.githubusercontent.com/9111558/51241553-0e3c7c80-19b9-11e9-857b-56c6931c9469.png)
- [ ] Ensure that the associated footprints match the [official footprint library](https://github.com/kicad/kicad-footprints)
Standard SSOP-24
- [ ] If there are matching footprint PRs, provide link(s) as appropriate
None
- [ ] Check the output of the Travis automated check scripts - fix any errors as required
Python checklib.py scripts passes without errors.

I took the previous symbol from the ADuM450x series as an example and edited it as necessary, the automated tests all pass.

There are three NIC pins that should not be connected (according to the datasheet), I'm not sure what the policy in Kicad is but I left them out of the schematic to avoid users connecting them accidentally.

This is my first contribution to Kicad so any comment is more than welcome!

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/KiCad/kicad-symbols?pullRequest=1429) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/KiCad/kicad-symbols?pullRequest=1429) before we can accept your contribution.<br/><hr/>**Luca Della Vedova** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/KiCad/kicad-symbols?pullRequest=1429) it.</sub>

myfreescalewebpage commented:
Hi @luca-della-vedova , thanks for contributing,

A few comments I have during my review:

ADuM5410:
- [ ] GND1 pin should be stacked
- [ ] GNDISO pins should be stacked
- [ ] Pin 7 is not connected
- [ ] Pin 8, 17 are Not Connected and should be added to the symbol
- [ ] Not Connected pins are type Not Connected, invisible, placed on the outline border of the symbol
- [ ] Pin 14 is Power Output and should be moved to the top of the symbol, next to VDD2

ADuM5411:
- [ ] GND1 pin should be stacked
- [ ] GNDISO pins should be stacked
- [ ] Pin 8, 17 are Not Connected and should be added to the symbol
- [ ] Not Connected pins are type Not Connected, invisible, placed on the outline border of the symbol
- [ ] Pin 14 is Power Output and should be moved to the top of the symbol, next to VDD2

ADuM5412:
- [ ] GND1 pin should be stacked
- [ ] GNDISO pins should be stacked
- [ ] Pin 8, 17 are Not Connected and should be added to the symbol
- [ ] Not Connected pins are type Not Connected, invisible, placed on the outline border of the symbol
- [ ] Pin 14 is Power Output and should be moved to the top of the symbol, next to VDD2

For all devices:
- [ ] You can add 100mil clearance between VI/VO and the PDIS/VE2/VSEL group, example below (same width than the other ADuM device already in the lib which is also better for the consistency).
![capture](https://user-images.githubusercontent.com/16516760/51305176-72ebeb80-1a3a-11e9-8a45-d4c136484515.PNG)

Cheers,
Joel

luca-della-vedova commented:
Hi Joel,

Thanks for all the feedback! Hope I can use it in the future to make better submissions :).
I added all the unconnected pins, trying to follow the convention I saw in the other symbols, fixed the width, stacked the ground pins, moved the VDD and introduced the clearance.
I also fixed the VE1 being unconnected in the ADuM5410 which was quite a serious mistake so thanks for spotting it!

Here is a screenshot of the current version:

![image](https://user-images.githubusercontent.com/9111558/51308036-2923f500-1a7c-11e9-8df8-6a9a425b4dc6.png)

Is this better?

Cheers!
/Luca

myfreescalewebpage commented:
Hi Luca!
You're welcome, perfect fix! No more comment at all on the symbols, merging.
Thanks for your contribution, hope to see new symbols from you :)
Joel
